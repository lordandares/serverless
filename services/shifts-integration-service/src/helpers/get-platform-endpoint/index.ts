interface DeviceReceiver {
  deviceToken: string
  platform: string
}

export default async function getPlatformEndpoint(
  client: AWS.SNS,
  receiver: DeviceReceiver,
): Promise<AWS.SNS.CreateEndpointResponse | void> {
  const platformConfig = {
    android: {
      arn: process.env.AWS_SNS_ANDROID_ARN,
      service: process.env.AWS_SNS_ANDROID_SERVICE,
    },
    ios: {
      arn: process.env.AWS_SNS_IOS_ARN,
      service: process.env.AWS_SNS_IOS_SERVICE,
    },
  }

  const { deviceToken, platform } = receiver

  if (!deviceToken || !platform) return

  const config = platformConfig[platform]

  const params = {
    PlatformApplicationArn: config.arn,
    Token: deviceToken,
  }

  const endpoint = await client.createPlatformEndpoint(params).promise()

  return endpoint
}
