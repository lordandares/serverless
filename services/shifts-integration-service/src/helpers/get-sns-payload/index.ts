export default function getSnsPayload(params) {
  const { endpointArn, message, title, type: notificationType } = params

  return {
    Message: JSON.stringify({
      default: message,
      APNS: JSON.stringify({
        aps: {
          alert: message,
          'content-available': 1,
          sound: 'default',
        },
        notificationType,
      }),
      APNS_SANDBOX: JSON.stringify({
        aps: {
          alert: message,
          'content-available': 1,
          sound: 'default',
        },
        notificationType,
      }),
      GCM: JSON.stringify({
        data: {
          data: {
            notificationType,
          },
          message,
          title,
        },
      }),
    }),
    MessageStructure: 'json',
    TargetArn: endpointArn,
  }
}
