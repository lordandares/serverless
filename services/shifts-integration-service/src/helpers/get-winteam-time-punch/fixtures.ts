export const validAreaLocation = {
  _id: '5d5a3fabfc13ae07b7000000',
  application: '5d5a3fabfc13ae07b0000000',
  name: 'Valid Area Location',
  timezone: 'Australia/Melbourne',
}

export const validLegacyLocation = {
  _id: '5d5a3fabfc13ae07b7000001',
  name: 'Valid Legacy Location',
  geo: {
    coordinates: [-10, 10],
  },
  timezone: 'Australia/Melbourne',
}

export const standardUser = {
  _id: '5d5a3fabfc13ae07b7000002',
  email: 'valid-email-address',
  firstName: 'first-name',
  lastName: 'last-name',
  username: 'user-name',
}

export const winteamUser = {
  ...standardUser,
  plugins: {
    winteam: {
      options: {
        employeeId: 'wt-employee-id',
      },
    },
  },
}

export const activeShift = {
  application: '5d5a3fabfc13ae07b0000000',
  breaks: [],
  start: {
    gps: {
      geometry: [45, 60],
    },
    time: '2019-01-23T00:00:00.000Z',
  },
  user: standardUser._id,
}

export const activeShiftWithBreak = {
  application: '5d5a3fabfc13ae07b0000000',
  breaks: [
    {
      _id: 'break-id',
      end: {
        time: '2019-01-23T00:00:00.000Z',
      },
      start: {
        time: '2019-01-23T00:00:00.000Z',
      },
    },
  ],
  start: {
    gps: {
      geometry: [45, 60],
    },
    time: '2019-01-23T00:00:00.000Z',
  },
  user: standardUser._id,
}

export const endedShift = {
  application: '5d5a3fabfc13ae07b0000000',
  breaks: [],
  end: {
    gps: {
      geometry: [46, 61],
    },
    time: '2019-01-23T06:00:00.000Z',
  },
  start: {
    gps: {
      geometry: [45, 60],
    },
    time: '2019-01-23T00:00:00.000Z',
  },
  user: standardUser._id,
}
