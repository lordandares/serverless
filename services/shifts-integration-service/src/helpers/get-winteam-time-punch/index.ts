import { errors, mongo, schemas } from '@lighthouse/serverless-common'
import { getOr, last } from 'lodash/fp'
import * as moment from 'moment-timezone'
import * as querystring from 'querystring'

// 2018-08-03 16:00:00.000-0400
const DATE_FORMAT = 'YYYY-MM-DD HH:mm:ss.SSSZZ'

interface TimePunchEvent extends TimePunch {
  event: string
}

interface TimePunch {
  result: {
    CallLogId: string
    HoursLogID?: string
    JobDescription: string
    JobID?: string
    JobNumber?: string
    PunchStatus: string
    PunchTime: string
    StatusReason?: string
  }
  shift: schemas.ShiftSchema
  user: schemas.UserSchema
}

interface TimePunchResponse {
  GeoCoordinate?: {
    Latitude: number
    Longitude: number
  }
  EmployeeId?: string
  EmployeeNumber?: string
  JobId?: string
  JobNumber?: string
  PunchTime: string
}

export async function getWinTeamTimePunch(
  timePunchEvent: TimePunchEvent,
): Promise<TimePunchResponse | void> {
  const { event, shift, user } = timePunchEvent
  const { application: applicationId, location: locationId } = shift

  if (!event) {
    throw new errors.ValidationError({
      message: 'Missing event from TimePunch',
    })
  }

  const employeeReference = getEmployeeReference(user)

  if (!employeeReference) {
    console.info('GetWinTeamTimePunch :: Missing Employee Reference', {
      shift,
      user,
    })

    return
  }

  const shiftEntity = getShiftEntity(shift, event)
  const shiftTimestamp = getOr(false, 'time', shiftEntity)

  if (!shiftTimestamp) {
    throw new errors.ValidationError({
      message: 'Missing timestamp from TimePunch',
      data: {
        shiftEntity,
        shiftTimestamp,
      },
    })
  }

  const jobReference = await getJobReference(
    shiftEntity,
    applicationId,
    locationId,
  )

  const punchTime = moment.utc(shiftTimestamp).format(DATE_FORMAT)

  const payload: TimePunchResponse = {
    ...employeeReference,
    ...jobReference,
    PunchTime: punchTime,
  }

  // Append GeoCoordinates if they exist
  const gpsCoordinates = getOr([], 'gps.geometry.coordinates', shiftEntity)

  if (gpsCoordinates && gpsCoordinates.length === 2) {
    payload.GeoCoordinate = {
      Latitude: gpsCoordinates[1],
      Longitude: gpsCoordinates[0],
    }
  }

  return payload
}

export async function getWinTeamMessages(
  timePunch: TimePunch,
): Promise<string | undefined> {
  const { result, shift, user } = timePunch

  const { JobNumber } = result

  const employeeReference = getEmployeeReference(user)
  if (!employeeReference) {
    console.info('GetWinTeamMessages :: Missing Employee Reference', {
      shift,
      user,
    })

    return
  }

  if (!JobNumber) {
    console.info('GetWinTeamMessages :: Missing Job Reference', {
      result,
      shift,
      user,
    })

    return
  }

  const punchTime = moment.utc(shift.start.time).format(DATE_FORMAT)

  const params = {
    ...employeeReference,
    JobNumber,
    PunchTime: punchTime,
  }

  return querystring.stringify(params)
}

export function getEmployeeReference(user: schemas.UserSchema): object | void {
  const employeeId = getOr(false, 'plugins.winteam.options.employeeId', user)
  const employeeNumber = getOr(
    false,
    'plugins.winteam.options.employeeNumber',
    user,
  )

  if (employeeId || employeeNumber) {
    return { EmployeeNumber: employeeId || employeeNumber }
  }
}

export async function getJobReference(
  shiftEntity: schemas.ShiftEventSchema,
  applicationId: string,
  locationId?: string,
): Promise<object | void> {
  const areaLocationId = getOr(false, 'area.location.id', shiftEntity)

  // If within an area location / legacy location, attempt to find the job number
  if (areaLocationId || locationId) {
    const isLegacy = !areaLocationId

    const collectionName = isLegacy ? 'locations' : 'areas'

    const queryLocationId = isLegacy ? locationId : areaLocationId
    const query = {
      _id: new mongo.ObjectId(queryLocationId),
      application: new mongo.ObjectId(applicationId),
    }

    const collection = await mongo.getCollection(collectionName)
    const location = await collection.findOne(query)

    if (location) {
      const jobId = getOr(false, 'plugins.winteam.options.jobId', location)
      const jobNumber = getOr(
        false,
        'plugins.winteam.options.jobNumber',
        location,
      )

      if (jobId || jobNumber) {
        return { JobNumber: jobId || jobNumber }
      }
    }
  }
}

export function getShiftEntity(
  shift: schemas.ShiftSchema,
  event: string,
): object | void {
  switch (event) {
    case 'shift-break-end':
      return last<schemas.ShiftBreakSchema>(shift.breaks)!.end
    case 'shift-end':
      return shift.end
    case 'shift-break-start':
      return last<schemas.ShiftBreakSchema>(shift.breaks)!.start
    case 'shift-start':
      return shift.start
    default:
      return {}
  }
}
