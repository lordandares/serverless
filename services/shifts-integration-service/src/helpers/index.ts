export { default as getPlatformEndpoint } from './get-platform-endpoint'
export { default as getSnsPayload } from './get-sns-payload'
export {
  getWinTeamMessages,
  getWinTeamTimePunch,
} from './get-winteam-time-punch'
export { default as notifyUser } from './notify-user'
export { default as parseWinTeamMessages } from './parse-winteam-messages'
export { default as sendMessage } from './send-message'
export {
  default as validateEndpointAttributes,
} from './validate-endpoint-attributes'
