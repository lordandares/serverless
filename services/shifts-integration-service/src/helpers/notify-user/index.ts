import { errors, schemas } from '@lighthouse/serverless-common'
import * as AWS from 'aws-sdk'
import { getOr, isEmpty } from 'lodash/fp'

import { getPlatformEndpoint, sendMessage } from '../'

interface NotifyUserParams {
  message: string
  user: schemas.UserSchema
}

export default async function notifyUser(
  params: NotifyUserParams,
): Promise<void> {
  const { message, user } = params

  if (!message || !user) {
    throw new errors.ValidationError({
      data: {
        message,
        user,
      },
      message: 'notifyUser :: Missing required params',
    })
  }

  const deviceReceivers = getOr([], 'speakerbox.deviceReceivers', user)

  if (isEmpty(deviceReceivers)) return

  const sns = new AWS.SNS()

  const endpoints: AWS.SNS.CreateEndpointResponse[] = await Promise.all(
    deviceReceivers.map(
      async receiver => await getPlatformEndpoint(sns, receiver),
    ),
  )

  await sendMessage(sns, endpoints, params)
}
