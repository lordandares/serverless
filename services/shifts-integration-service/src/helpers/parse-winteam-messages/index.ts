import { mongo } from '@lighthouse/serverless-common'
import { getOr } from 'lodash/fp'

export default function parseWinTeamMessages(winteamResponse) {
  const messages = getOr([], 'Messages', winteamResponse)

  return messages.map(message => {
    const {
      _id = new mongo.ObjectId(),
      Id,
      IsGlobal,
      Message,
      Response1,
      Response2,
      Response1Id,
      Response2Id,
      PunchActionID,
    } = message

    return {
      _id,
      isGlobal: !!IsGlobal,
      message: Message,
      messageId: Id,
      punchActionId: PunchActionID,
      response1: Response1,
      response1Id: Response1Id,
      response2: Response2,
      response2Id: Response2Id,
    }
  })
}
