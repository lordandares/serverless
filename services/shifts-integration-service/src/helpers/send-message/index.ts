import { getSnsPayload, validateEndpointAttributes } from '../'

interface SendMessageParams {
  message: string
  title: string
  type: string
}

export default async function sendMessage(
  client: AWS.SNS,
  endpoints: AWS.SNS.CreateEndpointResponse[],
  params: SendMessageParams,
): Promise<object[]> {
  const result = await Promise.all(
    endpoints.map(async endpoint => {
      const { EndpointArn: endpointArn } = endpoint

      await validateEndpointAttributes(client, endpointArn)

      const payload = getSnsPayload({ endpointArn, ...params })

      console.info('sending message', params)

      const result = await client.publish(payload).promise()

      console.info('sent message', result)
      return result
    }),
  )

  return result
}
