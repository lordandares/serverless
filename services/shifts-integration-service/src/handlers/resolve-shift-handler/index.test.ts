jest.mock('../../helpers')
jest.mock('../../helpers/get-winteam-time-punch', () => ({
  getWinTeamMessages: jest.fn(),
}))

import { mongo } from '@lighthouse/serverless-common'
import { winteam } from '@lighthouse/serverless-integrations'

import { notifyUser, parseWinTeamMessages } from '../../helpers'
import { getWinTeamMessages } from '../../helpers/get-winteam-time-punch'
import resolveShiftHandler, { getStartShiftMessages } from './'

mongo.getCollection = jest.fn()
winteam.request = jest.fn()

describe('handlers:resolve-shift-handler', () => {
  beforeEach(() => jest.resetAllMocks())

  it('should error if the `shiftId` is missing', async () => {
    expect.assertions(1)

    const event = {
      shift: {
        _id: '',
      },
    }

    expect(resolveShiftHandler(event)).rejects.toThrowErrorMatchingSnapshot()
  })

  it('should update a shift status to resolved with messages', async () => {
    expect.assertions(5)

    const event = {
      shift: {
        _id: '5d1eb8a4fc13ae1f85000000',
      },
      user: {
        _id: 'user-id',
      },
    }

    const shiftMessages = [
      {
        _id: new mongo.ObjectId('5d4697a7e652cbb5b680a004'),
        isGlobal: false,
        message: 'test message',
        messageId: 1,
        punchActionId: 6,
        response1: 'Yes',
        response1Id: 0,
        response2: 'Also Yes',
        response2Id: 2,
      },
    ]

    const shiftResponseObject = {
      Messages: [
        {
          Id: 1,
          IsGlobal: 0,
          Message: 'test message',
          PunchActionID: 6,
          Response1: 'Yes',
          Response1Id: 0,
          Response2: 'Also Yes',
          Response2Id: 2,
        },
      ],
    }

    const mockFindOneAndUpdate = jest.fn().mockResolvedValue({
      value: {
        ...event.shift,
        messages: shiftMessages,
        status: 'resolved',
      },
    })

    getWinTeamMessages.mockResolvedValue(
      'EmployeeId=wt-valid-employee-id&JobId=wt-valid-job-id&PunchTime=2019-01-23%2000%3A00%3A00.000%2B0000',
    )
    winteam.request.mockResolvedValue(shiftResponseObject)
    mongo.getCollection.mockResolvedValue({
      findOneAndUpdate: mockFindOneAndUpdate,
    })

    parseWinTeamMessages.mockReturnValue(shiftMessages)

    const result = await resolveShiftHandler(event)

    expect(mongo.getCollection).toHaveBeenCalledTimes(1)
    expect(mongo.getCollection).toBeCalledWith('shifts')

    expect(mockFindOneAndUpdate).toHaveBeenCalledTimes(1)
    expect(mockFindOneAndUpdate).toBeCalledWith(
      { _id: new mongo.ObjectId(event.shift._id) },
      {
        $set: {
          messages: shiftMessages,
          status: 'resolved',
        },
      },
      { returnOriginal: false },
    )

    expect(result).toMatchSnapshot()
  })

  it('should update a shift status to resolved if shift messages could not be retrieved', async () => {
    expect.assertions(5)

    const event = {
      shift: {
        _id: '5d1eb8a4fc13ae1f85000000',
      },
      user: {
        _id: 'user-id',
      },
    }

    const mockFindOneAndUpdate = jest.fn().mockResolvedValue({
      value: {
        ...event.shift,
        messages: [],
        status: 'resolved',
      },
    })

    getWinTeamMessages.mockResolvedValue(
      'EmployeeId=wt-valid-employee-id&JobId=wt-valid-job-id&PunchTime=2019-01-23%2000%3A00%3A00.000%2B0000',
    )
    winteam.request.mockRejectedValue(new Error('ServerError'))
    mongo.getCollection.mockResolvedValue({
      findOneAndUpdate: mockFindOneAndUpdate,
    })

    const result = await resolveShiftHandler(event)

    expect(mongo.getCollection).toHaveBeenCalledTimes(1)
    expect(mongo.getCollection).toBeCalledWith('shifts')

    expect(mockFindOneAndUpdate).toHaveBeenCalledTimes(1)
    expect(mockFindOneAndUpdate).toBeCalledWith(
      { _id: new mongo.ObjectId(event.shift._id) },
      {
        $set: {
          messages: [],
          status: 'resolved',
        },
      },
      { returnOriginal: false },
    )

    expect(result).toMatchSnapshot()
  })

  it('should send a notification for shift with messages that need immediately actioned', async () => {
    expect.assertions(7)

    const event = {
      shift: {
        _id: '5d1eb8a4fc13ae1f85000000',
      },
      user: {
        _id: 'user-id',
      },
    }

    const shiftMessages = [
      {
        _id: new mongo.ObjectId('5d4697a7e652cbb5b680a001'),
        isGlobal: true,
        message: 'test message',
        messageId: 1,
        punchActionId: null,
        response1: 'Yes',
        response1Id: 0,
        response2: 'Also Yes',
        response2Id: 0,
      },
    ]

    const shiftResponseObject = {
      Messages: [
        {
          Id: 1,
          IsGlobal: 1,
          Message: 'test message',
          PunchActionID: null,
          Response1: 'Yes',
          Response1Id: 0,
          Response2: 'Also Yes',
          Response2Id: 0,
        },
      ],
    }

    const mockFindOneAndUpdate = jest.fn().mockResolvedValue({
      value: {
        ...event.shift,
        messages: shiftMessages,
        status: 'resolved',
      },
    })

    getWinTeamMessages.mockResolvedValue(
      'EmployeeId=wt-valid-employee-id&JobId=wt-valid-job-id&PunchTime=2019-01-23%2000%3A00%3A00.000%2B0000',
    )
    mongo.getCollection.mockResolvedValue({
      findOneAndUpdate: mockFindOneAndUpdate,
    })
    notifyUser.mockResolvedValue(null)
    parseWinTeamMessages.mockReturnValue(shiftMessages)
    winteam.request.mockResolvedValue(shiftResponseObject)

    const result = await resolveShiftHandler(event)

    expect(mongo.getCollection).toHaveBeenCalledTimes(1)
    expect(mongo.getCollection).toBeCalledWith('shifts')

    expect(mockFindOneAndUpdate).toHaveBeenCalledTimes(1)
    expect(mockFindOneAndUpdate).toBeCalledWith(
      { _id: new mongo.ObjectId(event.shift._id) },
      {
        $set: {
          messages: shiftMessages,
          status: 'resolved',
        },
      },
      { returnOriginal: false },
    )

    expect(notifyUser).toHaveBeenCalledTimes(1)
    expect(notifyUser).toHaveBeenCalledWith({
      message: 'Compliance Message(s) received.  Open Lighthouse for details',
      title: 'Compliance Message',
      type: 'shift-resolved-with-messages',
      user: {
        _id: 'user-id',
      },
    })

    expect(result).toMatchSnapshot()
  })

  it('should update a shift status to confirmed without messages', async () => {
    expect.assertions(6)

    const event = {
      shift: {
        _id: '5d1eb8a4fc13ae1f85000000',
      },
      user: {
        _id: 'user-id',
      },
    }

    const mockFindOneAndUpdate = jest.fn().mockResolvedValue({
      value: {
        ...event.shift,
        messages: [],
        status: 'resolved',
      },
    })

    getWinTeamMessages.mockResolvedValue(null)
    mongo.getCollection.mockResolvedValue({
      findOneAndUpdate: mockFindOneAndUpdate,
    })
    parseWinTeamMessages.mockReturnValue([])
    winteam.request.mockResolvedValue(null)

    const result = await resolveShiftHandler(event)

    expect(mongo.getCollection).toHaveBeenCalledTimes(1)
    expect(mongo.getCollection).toBeCalledWith('shifts')

    expect(mockFindOneAndUpdate).toHaveBeenCalledTimes(1)
    expect(mockFindOneAndUpdate).toBeCalledWith(
      { _id: new mongo.ObjectId(event.shift._id) },
      {
        $set: {
          messages: [],
          status: 'resolved',
        },
      },
      { returnOriginal: false },
    )

    expect(notifyUser).toHaveBeenCalledTimes(0)

    expect(result).toMatchSnapshot()
  })
})

describe('helpers :: getStartShiftMessages', () => {
  it('should return an empty array if messages are empty/null', () => {
    expect.assertions(1)

    const result = getStartShiftMessages(null)

    expect(result).toEqual([])
  })

  it('should return both global messages, and messages matching the start shift punchActionIds', () => {
    expect.assertions(2)

    const messages = [
      {
        _id: new mongo.ObjectId('5d4697a7e652cbb5b680a000'),
        isGlobal: false,
        message: 'test message',
        messageId: 1,
        punchActionId: 6,
        response1: 'Yes',
        response1Id: 0,
        response2: 'Also Yes',
        response2Id: 2,
      },
      {
        _id: new mongo.ObjectId('5d4697a7e652cbb5b680a001'),
        isGlobal: true,
        message: 'global message 1',
        messageId: 1,
        punchActionId: 6,
        response1: 'Yes',
        response1Id: 0,
        response2: 'Also Yes',
        response2Id: 2,
      },
      {
        _id: new mongo.ObjectId('5d4697a7e652cbb5b680a002'),
        isGlobal: false,
        message: 'start shift message',
        messageId: 1,
        punchActionId: 3,
        response1: 'Yes',
        response1Id: 0,
        response2: 'Also Yes',
        response2Id: 2,
      },
    ]

    const result = getStartShiftMessages(messages)

    expect(result).toHaveLength(2)
    expect(result).toMatchSnapshot()
  })
})
