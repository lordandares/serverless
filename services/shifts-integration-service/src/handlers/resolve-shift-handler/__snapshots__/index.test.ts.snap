// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handlers:resolve-shift-handler should error if the \`shiftId\` is missing 1`] = `"The Resource \\"shift\\" with ID of \\"Unknown\\" could not be found"`;

exports[`handlers:resolve-shift-handler should send a notification for shift with messages that need immediately actioned 1`] = `
Object {
  "shift": Object {
    "_id": "5d1eb8a4fc13ae1f85000000",
    "messages": Array [
      Object {
        "_id": "5d4697a7e652cbb5b680a001",
        "isGlobal": true,
        "message": "test message",
        "messageId": 1,
        "punchActionId": null,
        "response1": "Yes",
        "response1Id": 0,
        "response2": "Also Yes",
        "response2Id": 0,
      },
    ],
    "status": "resolved",
  },
  "user": Object {
    "_id": "user-id",
  },
}
`;

exports[`handlers:resolve-shift-handler should update a shift status to confirmed without messages 1`] = `
Object {
  "shift": Object {
    "_id": "5d1eb8a4fc13ae1f85000000",
    "messages": Array [],
    "status": "resolved",
  },
  "user": Object {
    "_id": "user-id",
  },
}
`;

exports[`handlers:resolve-shift-handler should update a shift status to resolved if shift messages could not be retrieved 1`] = `
Object {
  "shift": Object {
    "_id": "5d1eb8a4fc13ae1f85000000",
    "messages": Array [],
    "status": "resolved",
  },
  "user": Object {
    "_id": "user-id",
  },
}
`;

exports[`handlers:resolve-shift-handler should update a shift status to resolved with messages 1`] = `
Object {
  "shift": Object {
    "_id": "5d1eb8a4fc13ae1f85000000",
    "messages": Array [
      Object {
        "_id": "5d4697a7e652cbb5b680a004",
        "isGlobal": false,
        "message": "test message",
        "messageId": 1,
        "punchActionId": 6,
        "response1": "Yes",
        "response1Id": 0,
        "response2": "Also Yes",
        "response2Id": 2,
      },
    ],
    "status": "resolved",
  },
  "user": Object {
    "_id": "user-id",
  },
}
`;

exports[`helpers :: getStartShiftMessages should return both global messages, and messages matching the start shift punchActionIds 1`] = `
Array [
  Object {
    "_id": "5d4697a7e652cbb5b680a001",
    "isGlobal": true,
    "message": "global message 1",
    "messageId": 1,
    "punchActionId": 6,
    "response1": "Yes",
    "response1Id": 0,
    "response2": "Also Yes",
    "response2Id": 2,
  },
  Object {
    "_id": "5d4697a7e652cbb5b680a002",
    "isGlobal": false,
    "message": "start shift message",
    "messageId": 1,
    "punchActionId": 3,
    "response1": "Yes",
    "response1Id": 0,
    "response2": "Also Yes",
    "response2Id": 2,
  },
]
`;
